streams/src/main/java/org/apache/kafka/streams/kstream/Named.java

```diff
@@ -0,0 +1,94 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream;
+
+import org.apache.kafka.streams.errors.InvalidProcessorException;
+
+import java.util.Objects;
+
+public class Named implements NamedOperation<Named> {
+
+    private static final int MAX_NAME_LENGTH = 249;
+
+    protected String name;
+
+    protected Named(final String name) {
+        this.name = name;
+        if (name != null)
+            validate(name);
+    }
+
+    /**
+     * Create a Named instance with provided name.
+     *
+     * @param name  the processor name to be used. If {@code null} a default processor name will be generated.
+     * @return      A new {@link Named} instance configured with name
+     */
+    public static Named as(final String name) {
+        Objects.requireNonNull(name, "name can't be null");
+        return new Named(name);
+    }
+
+    @Override
+    public String name() {
+        return name;
+    }
+
+    @Override
+    public Named withName(final String name) {
+        Objects.requireNonNull(name, "name can't be null");
+        return new Named(name);
+    }
+
+
+    /**
+     * Validate the specified processor name. Because the processor name can be used to derive a state store name,
+     * the name is validate against the same rules than for a topic name.
+     * @param processorName the processor name to validate.
+     *
+     * @throws InvalidProcessorException if the processor name is invalid.
+     */
+    private static void validate(final String processorName) {
+        if (processorName.isEmpty())
+            throw new InvalidProcessorException("Processor name is illegal, it can't be empty");
+        if (processorName.equals(".") || processorName.equals(".."))
+            throw new InvalidProcessorException("Processor name cannot be \".\" or \"..\"");
+        if (processorName.length() > MAX_NAME_LENGTH)
+            throw new InvalidProcessorException("Processor name is illegal, it can't be longer than " + MAX_NAME_LENGTH +
+                    " characters, topic name: " + processorName);
+        if (!containsValidPattern(processorName))
+            throw new InvalidProcessorException("Processor name \"" + processorName + "\" is illegal, it contains a character other than " +
+                    "ASCII alphanumerics, '.', '_' and '-'");
+    }
+
+    /**
+     * Valid characters for Kafka topics are the ASCII alphanumerics, '.', '_', and '-'
+     */
+    private static boolean containsValidPattern(final String topic) {
```

Comment:
Similar as above. Can we reuse some existing code?

Reply:
As above, the method comes from https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/internals/Topic.java#L77.

If we are OK to create a dependency between this two classes we jut make this method public.